unic-locale-impl/src/extensions/unicode.rs

```diff
@@ -120,15 +127,22 @@ impl std::fmt::Display for UnicodeExtensionList {
 
         f.write_str("-u")?;
 
+        for attr in &self.attributes {
+            write!(f, "-{}", attr)?;
+        }
+
+        // This could be a global const
+        let type_true = vec!(parse_type("true").unwrap());
+
         for (k, t) in &self.keywords {
             write!(f, "-{}", k)?;
-            for v in t {
-                write!(f, "-{}", v)?;
-            }
-        }
 
-        for attr in &self.attributes {
-            write!(f, "-{}", attr)?;
+            // Any type == ["true"] is omitted for canonical form
+            if t != &type_true {
```

Comment:
you still seem to be stripping `true` while serializing rather than when parsing.

Reply:
Sorry, I hadn't pushed the latest version as I was waiting on #24. I have now pushed my latest commit, rebased on master.